add cmake support; minor changes for working on Linux; Updated README

Jiaolong opened the pull request:
Added CMakeLists;
Tested on Ubuntu 16.04.

seeta-wushuzhe commented on FaceDetection/CMakeLists.txt:
```diff
@@ -0,0 +1,59 @@
+option(USE_OPENMP      "Set to ON to build use openmp"  ON)
+
+# Use C++11
+set(CMAKE_CXX_STANDARD 11)
```
It seems that the flag "-std=c++11" is not automatically added, and I have to add it manually into `CMAKE_CXX_FLAGS`. The same problem exists for OpenMP. I was wondering why this happens.

NewProggie replied:
This is because he forgot to bind that c++11 as standard to an actual build target. CMake won't automagically know what and which target to compile with c++11 enabled. So something like this should do:

```
set_target_properties(seeta_facedet_lib PROPERTIES CXX_STANDARD 11)
set_target_properties(seeta_facedet_lib PROPERTIES CXX_STANDARD_REQUIRED ON) 
```

seeta-wushuzhe replied:
Yes, this works! Many thanks.

seeta-wushuzhe commented:
It is preferred not to add the images. Can you remove them from the PR?

Jiaolong commented:
Thanks for the review! I have removed the data folder. Regarding to the C++11 CMAKE_CXX_FLAGS issue, I am not sure what is the actual reason. It works just fine on my Ubuntu 16.04. My cmake version is 3.5.1, I also updated the cmake version in the CMakeLists.txt.

seeta-wushuzhe commented on FaceDetection/CMakeLists.txt:
```diff
@@ -1,17 +1,23 @@
+endif()
 
 # Use OpenMP
 if (USE_OPENMP)
```
It seems that `OpenMP_FOUND` on line 25 should be `OPENMP_FOUND`. The former works fine for CMake 3.5 or newer, while the latter can also be used with CMake 3.1.

seeta-wushuzhe commented on FaceDetection/CMakeLists.txt:
```diff
@@ -1,17 +1,23 @@
-cmake_minimum_required(VERSION 2.8.4)
+cmake_minimum_required(VERSION 3.5.1)
```
CMake 3.1.0 or newer is OK for everything except for `OpenMP_FOUND`. So maybe we can use `OPENMP_FOUND` and `cmake_minimum_required(VERSION 3.1.0)`?

Jiaolong commented:
@NewProggie  Thanks for checking the cmake version issue.
@seeta-wushuzhe I have changed the minimum version to 3.1.0  as well as OPENMP_FOUND, thanks for the review.

seeta-wushuzhe commented:
@Jiaolong Thanks for your effort and prompt reply! I think everything is fine now.
